test/kitchen_sink/map_test.exs

```diff
@@ -38,6 +38,12 @@ defmodule KitchenSink.MapTest do
     assert KMap.deep_merge([%{a: %{b: %{c: %{d: 1}}}}, %{a: %{z: 1}}]) == %{a: %{b: %{c: %{d: 1}}, z: 1}}
   end
 
+  test "map with list" do
+    assert KMap.deep_merge(%{a: [%{b: 1}, %{c: 2}]}, %{a: [%{d: 3}, %{e: 4}]}) == %{a: [%{b: 1, d: 3}, %{c: 2, e: 4}]}
```

Comment:
i don't see tests for empty lists or nil or stuff that will probably break programs depending on this feature. i think it may be a lot more complicated that first thought...